code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -173,14 +173,23 @@
 
 	//Fill machine with a bag!
 	if(istype(I, /obj/item/storage/bag))
+		if(!I.contents.len)
+			to_chat(user, span_notice("[I] is empty!"))
+			return TRUE
+
 		var/list/inserted = list()
 		if(I.atom_storage.remove_type(/obj/item/food/grown, src, limit - length(holdingitems), TRUE, FALSE, user, inserted))
 			for(var/i in inserted)
 				holdingitems[i] = TRUE
-			if(!I.contents.len)
-				to_chat(user, span_notice("You empty [I] into [src]."))
-			else
-				to_chat(user, span_notice("You fill [src] to the brim."))
+		inserted = list()
```

Comment:
```suggestion
			inserted = list()
```

I would rather we only clear the list if we add stuff into it, because otherwise we're just doing useless work.

Reply:
that's fair, I'll change it.

Reply:
I didn't realize I could just commit your changes? that's actually awesome, anyway I fixed it, the tests are running again right now.